Modules/Core/Common/src/itkProcessObject.cxx

```diff
@@ -90,6 +90,7 @@ ::MakeOutput( const DataObjectIdentifierType & name )
     {
     return this->MakeOutput( this->MakeIndexFromOutputName(name) );
     }
+  // DataObject::New() returns is from Object, and returns only Object!
```

Comment:
itkDataObject.h is missing `itkTypeMacro`.

Reply:
Oops, it has that but it is missing definition of new.

Reply:
`Object` has manually defined new, whereas `DataObject` has neither `itkNewMacro` nor manually defined/overridden `New`.

Reply:
Perhaps both `Object` and `DataObject` should use `itkNewMacro`?

Reply:
Yup. The questions is: Should` DataObject` be instantiable or an abstract base class?

Reply:
Is `DataObject` currently instantiable? It looks like it is. So just fix the way it is instantiated? If I am wrong, what are the abstract members?

Reply:
New method for `DataObject` seems to work just fine. Re: `Object::New` likely has to be in the cxx file due to a circular dependency with using the `ObjectFactory`